controller/Controller.php

```diff
@@ -517,4 +515,107 @@ function passwordPage($id)
+            if ($this->_val->validateItem($_POST["itemName"])) {
+                $this->_db->updateItem($_POST["itemId"], $_POST["itemName"], $_POST["itemType"]);
+            } else {
+                $this->_f3->set("errors", $this->_val->getErrors());
+            }
+
+            if ($itemId == 0) {
+                if ($this->_val->validateItem($_POST["itemName"])) {
+                    $itemId = $this->_db->addItem($_POST["itemName"], $_POST["itemType"]);
+                    $this->_f3->reroute("edit/$itemId");
+                }
+            }
+        }
+
+        // Save State
+        if (isset($_POST["stateSave"])) {
+
+            // Changing isDone to an int
+            if (!isset($_POST["stateIsDone"])) {
+                $_POST["stateIsDone"] = 0;
+            } else {
+                $_POST["stateIsDone"] = 1;
+            }
+
+            // Update existing state
+            if ($_POST["stateId"] != 0) {
+                if ($this->_val->validateState($_POST["stateId"], $_POST["stateName"], $_POST["stateText"])) {
+                    $this->_db->updateState($_POST["stateId"], $_POST["stateName"], $_POST["stateSetBy"], $_POST["stateText"], $_POST["stateIsDone"]);
+                } else {
+                    $this->_f3->set("errors", $this->_val->getErrors());
+                }
+                // Add new state
+            } else {
+                if ($this->_val->validateState($_POST["stateId"], $_POST["stateName"], $_POST["stateText"])) {
+                    $this->_db->addState($itemId, $_POST["stateName"], $_POST["stateSetBy"], $_POST["stateText"], $_POST["stateIsDone"]);
+                } else {
+                    $this->_f3->set("errors", $this->_val->getErrors());
+                    $this->_f3->set("stateNameNew", $_POST["stateName"]);
+                    $this->_f3->set("stateSetByNew", $_POST["stateSetBy"]);
+                    $this->_f3->set("stateTextNew", $_POST["stateText"]);
+                    $this->_f3->set("stateIsDoneNew", $_POST["stateIsDone"]);
+                }
+            }
+
+        }
+
+        // Move Up
+        if (isset($_POST["moveUp"])) {
+            $this->_db->updateStateOrder($_POST["stateId"], -1);
+        }
+
+        // Move Down
+        if (isset($_POST["moveDown"])) {
+            $this->_db->updateStateOrder($_POST["stateId"], 1);
+        }
+
+        // Check for default state warnings
+        $defaults = $this->_db->getStateCount($itemId, "default");
+        if ($defaults > 1) {
+            $this->_f3->set("defaultWarning", "You have more than one default state set! Please have only one default state for this item. Having more than one default states can result in errors displaying the item.");
+        } else if ($defaults < 1) {
+            $this->_f3->set("defaultWarning", "You do not have a default state set! Please have exactly one default state for this item. Having no default states can result in errors displaying the item.");
+        }
+
```

Comment:
Maybe just some inline comments about what these if statements do. (For if statement below)